packages/frontend/src/redux/slices/transactions/index.js

```diff
@@ -1,22 +1,149 @@
+    async ({ hash, signer_id, accountId }, { dispatch, getState }) => {
+        let status;
+        try {
+            const transactionDetails = await transactionExtraInfo({ hash, signer_id });
+            status = Object.keys(transactionDetails.status)[0];
+        } catch (error) {
+            status = 'notAvailable';
+        }
+        const checkStatus = ['SuccessValue', 'Failure'].includes(status);
+        const { actions: { updateTransactionStatus } } = transactionsSlice;
+        dispatch(updateTransactionStatus({ status, checkStatus, accountId, hash }));
+    }
+);
+
+const transactionsSlice = createSlice({
+    name: SLICE_NAME,
+    initialState,
+    reducers: {
+        setTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+            set(state, ['transactions', 'byAccountId', accountId, 'items'], transactions);
+        },
+        updateTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+
+            const transactionsState = state.transactions.byAccountId[accountId].items;
+            const hash = transactionsState.map((t) => t.hash_with_index);
+
+            // when updating the transaction, we do not want to replace the entire array, because for some entries the tx status may already be fetched
+            transactions.forEach((t) => {
+                if (!hash.includes(t.hash_with_index)) {
+                    transactionsState.unshift(t);
+                    if (transactionsState.length > 10) {
+                        transactionsState.pop();
+                    }
+                }
+            });
+        },
+        updateTransactionStatus(state, { payload }) {
+            const { status, checkStatus, accountId, hash } = payload;
+
+            const transactionsState = state.transactions.byAccountId[accountId].items;
+
+            const transactionEntry = transactionsState.find((t) => t.hash === hash);
+            if (transactionEntry) {
+                Object.assign(transactionEntry, { status, checkStatus});
+            }
+        }
+    },
+    extraReducers: ((builder) => {
+        builder.addCase(fetchTransactions.pending, (state, { meta }) => {
+            const { accountId } = meta.arg;
+
+            set(state, ['transactions', 'byAccountId', accountId, 'status', 'loading'], true);
+        });
+        builder.addCase(fetchTransactions.fulfilled, (state,  { meta }) => {
+            const { accountId } = meta.arg;
+
+            set(state, ['transactions', 'byAccountId', accountId, 'status', 'loading'], false);
+        });
+        builder.addCase(fetchTransactions.rejected, (state, { meta, error }) => {
+            const { accountId } = meta.arg;
+            
+            set(state, ['transactions', 'byAccountId', accountId, 'status', 'loading'], false);
+            set(state, ['transactions', 'byAccountId', accountId, 'status', 'error'], error?.message || 'An error was encountered.');
```

Comment:
I think we should store error code?

Reply:
That sounds sane.  Maybe `error` would instead be an object with either (or both) `code` and `message` properties?

Not blocking this PR, but we need some documentation about this pattern written down, so that there is a clear path for everyone to follow as we implement reducers that track error states without using the global alert reducer.

Can you create an issue to commit a markdown file in the `src/redux` directory that outlines this and follow up?  Since you are the most familiar with our current error handling, I'm looking for you to have a think about what we need, and what the data structure should look like.

We also need a basic description outlining that we use `redux-toolkit-js` slices for reducer logic (link to package documentation), reselect selectors for all data accesses (link to package documentation). In a nutshell, we need some developer documentation around Redux patterns in place, since not all of our code follows our current best practices and so there is no obvious golden path just by looking at our existing code usages.

Reply:
Done. Issue added.

Reply:
> That sounds sane. Maybe error would instead be an object with either (or both) code and message properties?

I will add the same change for other slices in separate PR.